api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -645,27 +648,47 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     }
     dbWorkspace.setReviewRequested(researchPurpose.getReviewRequested());
 
-    // Clone the previous description, by default.
+    // Clone description/CDR version from the source, by default.
     if (Strings.isNullOrEmpty(toWorkspace.getDescription())) {
       dbWorkspace.setDescription(fromWorkspace.getDescription());
     } else {
       dbWorkspace.setDescription(toWorkspace.getDescription());
     }
-
-    dbWorkspace.setCdrVersion(fromWorkspace.getCdrVersion());
-    dbWorkspace.setDataAccessLevel(fromWorkspace.getDataAccessLevel());
-
-    WorkspaceUserRole permissions = new WorkspaceUserRole();
-    permissions.setRoleEnum(WorkspaceAccessLevel.OWNER);
-    permissions.setWorkspace(dbWorkspace);
-    permissions.setUser(user);
-
-    dbWorkspace.addWorkspaceUserRole(permissions);
-
-    dbWorkspace = workspaceService.saveAndCloneCohortsAndConceptSets(fromWorkspace, dbWorkspace);
-    CloneWorkspaceResponse resp = new CloneWorkspaceResponse();
-    resp.setWorkspace(TO_SINGLE_CLIENT_WORKSPACE_FROM_FC_AND_DB.apply(dbWorkspace, toFcWorkspace));
-    return ResponseEntity.ok(resp);
+    String reqCdrVersionId = body.getWorkspace().getCdrVersionId();
+    if (Strings.isNullOrEmpty(reqCdrVersionId) ||
+        reqCdrVersionId.equals(Long.toString(fromWorkspace.getCdrVersion().getCdrVersionId()))) {
+      dbWorkspace.setCdrVersion(fromWorkspace.getCdrVersion());
+      dbWorkspace.setDataAccessLevel(fromWorkspace.getDataAccessLevel());
+    } else {
+      CdrVersion reqCdrVersion = setCdrVersionId(dbWorkspace, reqCdrVersionId);
+      dbWorkspace.setDataAccessLevelEnum(reqCdrVersion.getDataAccessLevelEnum());
+    }
+
+    WorkspaceUserRole ownerRole = new WorkspaceUserRole();
+    ownerRole.setRoleEnum(WorkspaceAccessLevel.OWNER);
+    ownerRole.setWorkspace(dbWorkspace);
+    ownerRole.setUser(user);
+
+    dbWorkspace.addWorkspaceUserRole(ownerRole);
+    org.pmiops.workbench.db.model.Workspace savedWorkspace =
+        workspaceService.saveAndCloneCohortsAndConceptSets(fromWorkspace, dbWorkspace);
```

Comment:
It'd be nice if this could be done in a Transactional block with the workspace creation, just to make sure everything gets committed at once... but I can't remember if that actually doesn't work for some reason. Anyway, not a huge deal.